Source/Plugins/Core/com.equella.core/src/com/tle/web/searching/section/SearchQuerySection.java

```diff
@@ -623,6 +623,22 @@ private void doBasicSearch(SectionInfo info, String query) {
     searchResults.startSearch(info);
   }
 
+  public static void basicSearchForNewSearch(
+      SectionInfo from, String query, boolean inSelectOrAdd) {
+    // In selectOrAdd the endpoint is '/selectoradd/searching.do'.
+    SectionInfo info =
+        inSelectOrAdd
+            ? from.createForward(SearchSelectable.NEW_FORWARD_PATH)
+            : RootSearchSection.createForward(from);
+    String basicUrl = info.getPublicBookmark().getHref();
+
+    String queryString = ((InfoBookmark) info.getPublicBookmark()).getQuery();
+    if (queryString.isEmpty()) {
+      basicUrl = basicUrl + "?";
+    }
+    from.forwardToUrl(basicUrl + "&q=" + query);
```

Comment:
As discussed, probably want to rework to this rough idea - maybe need try/catch too.

```suggestion
    URL basicUrl = new URL(info.getPublicBookmark().getHref());
    
    from.forwardToURL(
      basicUrl.toString()
      + (basicUrl.getQuery().isEmpty ? "?" : "&")
      + "q=" + query));
```

Reply:
Oops.... URL.getQuery() returns `null` if there is no query, so that should be `basicUrl.getQuery() == null`.